lib/widgets/board_icon.dart

```diff
@@ -8,10 +9,13 @@ import '../model/game.dart';
 /// It holds board value named [board].
 class BoardIconView extends StatelessWidget {
   const BoardIconView({super.key, required this.board});
+
   final Board board;
 
   @override
   Widget build(BuildContext context) {
+    final int boardLength = board.length;
```

Comment:
Why length?
What is possible is hard to detect programmatically, because board may contain holes. 
So let's for now just allow to select 2..10 in-a-row. And it is ok if it is not possible.

Reply:
done